agent/background.go

```diff
@@ -33,9 +36,39 @@ func (a *Agent) startHeartbeat(ctx context.Context, preAndMainCancel context.Can
 	defer ticker.Stop()
 
 	var hasSentAbort bool
+	var loggedTimeout bool
 	for {
 		select {
 		case <-ticker.C:
+			if tc.hadHeartbeatTimeout() {
+				// If the heartbeat hits its maximum reasonable timeout, that
+				// means something unrecoverably wrong has most likely occurred
+				// in the task runtime (e.g. a deadlock in the agent has stalled
+				// the task).
+				timeoutOpts := tc.getHeartbeatTimeout()
+				timeout := timeoutOpts.getTimeout()
+				if !loggedTimeout {
+					msg := fmt.Sprintf("Heartbeat has hit maximum allowed '%s' timeout of %s, task is at risk of timing out if it runs for much longer.", timeoutOpts.kind, timeout.String())
+					grip.Alert(message.Fields{
```

Comment:
It's not necessary, but it would make filtering the Splunk logs slightly easier. The message includes this info solely because it's going to be for the execution log sent to the user in the follow-up PR.